src/Accelerators/InitAccelerators.cpp

```diff
@@ -0,0 +1,34 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===------------------------ InitAccelerators.cpp ------------------------===//
+//
+// Copyright 2022 The IBM Research Authors.
+//
+// =============================================================================
+//
+// Initialization of accelerators.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Accelerators/Accelerator.hpp"
+
+namespace onnx_mlir {
+namespace accel {
+
+#ifdef HAS_ACCELERATORS
```

Comment:
You don't need the ifdef here because if it's not defined, the macro below will expand to nothing anyway. Because the init function needs to return a value depending on the existence of accelerators, it needs the ifdef. I haven't looked at the code, but maybe init can return void since if there are no accelerators, init would succeed by default? In that case you could drop the ifdef there as well.

Reply:
Ok, I looked at the usage and sometimes the output from `initAccelerators` is used to determine if there are any and sometimes it's not. I would recommend changing `initAccelerators` to return void, removing the need for `HAS_ACCELERATORS` and then modifying the code that was using `initAccelerators` to check for accelerators, to just do it the same way as other places and use the output of `getAccelerators`.

Reply:
Done.